src/ucs/datastruct/bitmap.h

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_BITMAP_H_
+#define UCS_BITMAP_H_
+
+#include <ucs/sys/compiler_def.h>
+#include <stdint.h>
+
+BEGIN_C_DECLS
+
+
+#define _UCS_BITMAP_BITS_IN_WORD (64)
+
+
+#define _UCS_BITMAP_WORD_INDEX(_bit_index) ((_bit_index) / _UCS_BITMAP_BITS_IN_WORD)
+
+
+#define _UCS_BITMAP_BIT_IN_WORD_INDEX(_bit_index) ((_bit_index) % _UCS_BITMAP_BITS_IN_WORD)
+
+/*
+ * Represents an n-bit bitmap, where n is a multiple of 64, by using an array
+ * of 64-bit unsigned long integers.
+ * 
+ * @param _length Bits number in the bitmap
+ */
+#define _UCS_BITMAP_DECLARE_TYPE(_length) \
+    typedef struct { \
+        uint64_t bits[_UCS_BITMAP_WORD_INDEX(_length)]; \
+    } ucs_bitmap_t(_length);
+
+
+/**
+ * Expands to bitmap type definition
+ *
+ * @param _length Bits number (as passed to UCS_BITMAP_DECLARE_TYPE)
+ *
+ * Example:
+ *
+ * @code{.c}
+ * ucs_bitmap_t(64) my_bitmap;
+ * @endcode
+ */
+#define ucs_bitmap_t(_length) \
+    _UCS_BITMAP_IDENTIFIER(_length, _t)
+
+
+#define _UCS_BITMAP_IDENTIFIER(_name, _suffix) \
+    UCS_PP_TOKENPASTE(ucs_bitmap_, UCS_PP_TOKENPASTE(_name, _suffix))
+
+
+_UCS_BITMAP_DECLARE_TYPE(64)
+_UCS_BITMAP_DECLARE_TYPE(128)
+_UCS_BITMAP_DECLARE_TYPE(256)
```

Comment:
why do we need to declare those types? IMO can leave it to users

Reply:
Types are identified only by bits number, and if we leave it to users - several types with the same name (e.g `ucs_bitmap_128_t`) might be declared by mistake, which would lead to conflicts.